src/Common/src/CoreLib/System/Version.cs

```diff
@@ -405,14 +406,20 @@ private static bool TryParseComponent(ReadOnlySpan<char> component, string compo
             return int.TryParse(component, NumberStyles.Integer, CultureInfo.InvariantCulture, out parsedComponent) && parsedComponent >= 0;
         }
 
+        // Force inline as the true/false ternary takes it above ALWAYS_INLINE size even though the asm ends up smaller
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool operator ==(Version v1, Version v2)
         {
-            if (v1 is null)
+            // Test "right" first to allow branch elimination when inlined for null checks (== null)
+            // so it can become a simple test
+            if (v2 is null)
             {
-                return v2 is null;
+                // return true/false not the test result https://github.com/dotnet/coreclr/issues/914
+                return (v1 is null) ? true : false;
```

Comment:
> return (v1 is null) ? true : false; [](start = 15, length = 36)

this can be written as return v1 == null

Reply:
Is working around a Jit codegen inefficiency https://github.com/dotnet/coreclr/issues/914